### PR TITLE
Missing Business Flow Execution Status in Execution Logs on Exception

### PR DESCRIPTION
**Pull request by vyasprateek**
Issue Fixed on handling exception. Need to confirm the Catch Exception Error text

### CODE REVIEW

File: Ginger/GingerCoreNET/Run/RunListenerLib/ExecutionLoggerManager.cs [hunk 1]
```diff
@@ -322,21 +322,27 @@ public override void BusinessFlowStart(uint eventTime, BusinessFlow businessFlow
...
         {
             mContext.BusinessFlow = businessFlow;
-            Object BFR = mExecutionLogger.SetReportBusinessFlow(mContext, offlineMode, ExecutedFrom, this.Configuration.ExecutionLoggerConfigurationIsEnabled);
+            Object BFR = null;
 
-            if (this.Configuration.ExecutionLoggerConfigurationIsEnabled)
+            try
             {
-                if (this.ExecutedFrom == Amdocs.Ginger.Common.eExecutedFrom.Automation)
+                BFR = mExecutionLogger.SetReportBusinessFlow(mContext, offlineMode, ExecutedFrom, this.Configuration.ExecutionLoggerConfigurationIsEnabled);
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/RunListenerLib/ExecutionLoggerManager.cs:329]: which peace of code was failing? mExecutionLogger.SetReportBusinessFlow? so BFR won't be Null again? let's talk on it on next meeting.

File: Ginger/GingerCoreNET/Run/RunListenerLib/ExecutionLoggerManager.cs [hunk 2]
```diff
@@ -322,21 +322,27 @@ public override void BusinessFlowStart(uint eventTime, BusinessFlow businessFlow
...
+                if (this.Configuration.ExecutionLoggerConfigurationIsEnabled)
                 {
-                    this.ExecutionLogBusinessFlowsCounter = 0;
-                    mExecutionLogger.ExecutionLogBusinessFlowsCounter = 0;
-                    this.BFCounter = 0;
+                    if (this.ExecutedFrom == Amdocs.Ginger.Common.eExecutedFrom.Automation)
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/Run/RunListenerLib/ExecutionLoggerManager.cs:332]: did you tested the changes also for regular executions on Automate and Run set to make sure Report comes as expected?

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/GingerCoreNET/Run/RunListenerLib/TextFileRepository.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6313097&bid=20526072)

**Review by MeniKadosh1 — APPROVED** [review]

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Ginger/GingerCoreNET/Run/RunListenerLib/TextFileRepository.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6313097&bid=20526072)

**Pull request merged**